crates/driver/src/domain/competition/order/mod.rs

```diff
@@ -18,7 +18,13 @@ pub struct Order {
     /// The user specified a custom address to receive the output of this order.
     pub receiver: Option<eth::Address>,
     pub valid_to: util::Timestamp,
+    /// The amount that this order ideally wants to sell. When the order is
+    /// fulfilled, the actual sold amount isn't always exactly equal to this
+    /// amount, e.g. because the order was partial or due to slippage.
     pub sell: eth::Asset,
+    /// The amount that this order ideally wants to buy. When the order is
+    /// fulfilled, the actual bought amount isn't always exactly equal to this
+    /// amount, e.g. because the order was partial or due to slippage.
     pub buy: eth::Asset,
```

Comment:
"Ideally wants to sell/buy" seems like an odd choice of words. Ideally you would for example get way more than the specified `buy` amount in case this order is a sell order. Maybe it makes sense to say:
```rust
/// The maximum amount this order is allowed to sell.
/// The final executed amount depends on partial fills and the order side.
pub sell: eth::Asset,
/// The minimum amount this order must buy.
/// The final executed amount depends on partial fills and the order side.
pub sell: eth::Asset,
```
It's tricky to come up with a concise but accurate description because a few different factors are at play here.